File: HackerRank/JavaSolutions/src/com/nkborba/mergeStrings.java
package com.nkborba;

public class mergeStrings {

    public String merge(String a, String b){

        String result = "";

        for(int cont = 0; cont < a.length() || cont < b.length(); cont++){
            if(cont < a.length()){
                result += a.charAt(cont);
            }
            if(cont < b.length()){
                result += b.charAt(cont);
            }
        }

        return result;
    }

}

File: HackerRank/kotlinSolutions/src/mergeStrings.kt
class MergeStrings {

    fun merge(a: String, b: String): String {
        var result = ""
        var larger = if (a.length > b.length) a.length else b.length

        for (cont: Int in 0 until larger) {
            if (cont < a.length) result += a[cont]
            if (cont < b.length) result += b[cont]
        }

        return result
    }
}